### ISSUE TITLE
WITCH ON THE HOLY NIGHT (2052410)

### ISSUE THREAD

**Issue by MuraPura** [ValveSoftware/Proton#7348]
# Compatibility Report
- Name of the game with compatibility issues: WITCH ON THE HOLY NIGHT
- Steam AppID of the game: 2052410

## System Information
- GPU: Radeon RX 6950 XT
- Video driver version: latest mesa-git
- Kernel version: 6.6.7-arch1-1 
- Link to full system information report as [Gist](https://gist.github.com/): https://gist.github.com/MuraPura/9793117765d7a13f8f6a6a5a2f2d3e8f
- Proton version: Proton 8.0-4 and Proton Experimental

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-2052410 (Experimental).log](https://github.com/ValveSoftware/Proton/files/13675819/steam-2052410.Experimental.log)

[steam-2052410 (8.0-4).log](https://github.com/ValveSoftware/Proton/files/13675821/steam-2052410.8.0-4.log)

## Symptoms
Crashes after booting, cannot get past the loading screen.

## Reproduction
Launch the game, no other steps necessary.

Please note: some Steam Deck users are reporting that Proton 6.3-8 works. I am unable to produce this on Desktop Linux, game will not boot at all on that version for me.

**Comment by matte-schwartz** [comment]
I was one of the users who commented that Proton 6.3-8 was more stable, but even that eventually crashes on my Steam Deck as well. the easiest test is to fast forward into the middle of Chapter 1 when the intro video sequence should start to play. I can hear background rain noises and soundtrack but the screen stays blank. Here is as copy of my log with 6.3-8: [steam-2052410.log](https://github.com/ValveSoftware/Proton/files/13676627/steam-2052410.log)

**Comment by WillTer** [comment]
I can play a game on Proton Experimental and with PROTON_USE_WINED3D=1 (tested on Steam Deck and Linux PC with Fedora 39), but it still crashes after some random time.
Log from my PC: [steam-2052410.log](https://github.com/ValveSoftware/Proton/files/13676713/steam-2052410.log)

**Comment by R1kaB3rN** [comment]
Unfortunately, this game is basically borked at the moment and I hope the developers look into this and provide an update. The game can crash after pressing "Y" then "B", continuously pressing "A" to progress the script or pressing "R2" to fast-forward anywhere in the intro.

Proton version: **experimental-bleeding-edge-8.0-69287-20231214-pf2ab99-w1104c5-d1b31aa-v06bf46**
OS: **SteamOS (3.5.11)**
Proton Log: https://gist.github.com/R1kaB3rN/4b356563c871a645527b3dcbb9141ab7

**Comment by R1kaB3rN** [comment]
This _might_ be a regression in Proton because using 7.0-6 the crashes seem to no longer occur in my attempt in reproducing them, and below I listed a log. @MuraPura do you mind confirming this?

Proton version: 7.0-6
OS: SteamOS (3.5.11)
Proton Log: https://gist.github.com/R1kaB3rN/e0442ebeda160d7d8f0550e3826f584e

**Comment by matte-schwartz** [comment]
On that version and older you can make it to the intro video that starts playing during the rain sequence, but it will skip that entire intro and dump you when Aoko arrives at the school after the intro finished.

**Comment by R1kaB3rN** [comment]
> On that version and older you can make it to the intro video that starts playing during the rain sequence, but it will skip that entire intro and dump you when Aoko arrives at the school after the intro finished.

Downgrading, though not ideal, suffices for now. The main showstopper was the crashes which made the game completely unplayable. But even then, I haven't played long enough yet to conclude that it still will not crash

**Comment by MuraPura** [comment]
> This _might_ be a regression in Proton because using 7.0-6 the crashes seem to no longer occur in my attempt in reproducing them, and below I listed a log. @MuraPura do you mind confirming this?
> 
> Proton version: 7.0-6 OS: SteamOS (3.5.11) Proton Log: https://gist.github.com/R1kaB3rN/e0442ebeda160d7d8f0550e3826f584e

I cannot confirm this - the game will not even boot on my system with 7.0-6, it will not even generate a log. The way this game behaves with Proton seems to differ between Steam Deck and desktop based on what I am seeing.

**Comment by WillTer** [comment]
Switching to 7.0-6 makes me possible to finish the entire chapter 2 without any crashes. Still using PROTON_USE_WINED3D, because it seems to be more stable on the other versions of proton with it.

**Comment by uunniiblog** [comment]
Requires a bit of workaround to get to it but I've got it to work in stable on both desktop and steam deck: 
Need to use D3D with proton version 6.3-8 to avoid potential crashes and install some video codecs to play the videos ingame (One plays at the logo boot, another after the phone ringing at the start, and I'd guess the ending credits too but haven't reached that part)

1. Get `<Workaround removed by moderator>` and unzip it
2. In Steam, in compatibility select Proton 6.3-8
3. In launch options add this to enable D3D: PROTON_USE_WINED3D=1 %command%
4. open the game in steam once to create a wineprefix
5. open protontricks -> select the game -> default wineprefix -> Run a commandline shell (for debugging) 
6. in the terminal that opens run: `<Workaround removed by moderator>` 
(or the folder where you unzipped it)
7. Close the terminal and protontricks after its finished and play the game

I've tested it with a few versions of proton 8+ but they seem to crash randomly. With 6.3-8 seems stable haven't had it crash at least skipping around and the videos work thanks to the codec script (it's a modified version of mf with a few extra dlls)

**Comment by kisak-valve** [comment]
Hello @uunniiblog, the workaround you posted is legally problematic and has been removed.

**Comment by YoreonX** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7348#issuecomment-1858844435

This workaround would be more useful in-Steam Discussion threads about this game specifically or ProtonDB.

This thread is to highlight comatability issues so Proton Developers can research it, seeing as the game is currently untested atm.

**Comment by gofman** [comment]
This should hopefully work better in the just updated Proton Experimental ([bleeding-edge] beta branch) (without any workarounds).

**Comment by matte-schwartz** [comment]
i've just updated to Bleeding Edge, cleared out my Proton prefix, and tried again. sometimes it crashes right at launch, sometimes you can make it to the main menu where it crashes once you press Start. Here is my Proton log when the latter happened: 
[steam-2052410.log](https://github.com/ValveSoftware/Proton/files/13710369/steam-2052410.log)

**Comment by matte-schwartz** [comment]
`PROTON_USE_WINED3D=1 %command%` as a launch option will get you to the line about cats in the intro until it crashes. 
[steam-2052410.log](https://github.com/ValveSoftware/Proton/files/13710392/steam-2052410.log)

**Comment by gofman** [comment]
> i've just updated to Bleeding Edge, cleared out my Proton prefix, and tried again. sometimes it crashes right at launch, sometimes you can make it to the main menu where it crashes once you press Start. Here is my Proton log when the latter happened: [steam-2052410.log](https://github.com/ValveSoftware/Proton/files/13710369/steam-2052410.log)

From your log, you are not using the latest [bleeding-edge]. If you had Steam running, it might worth restarting it to pick the just updated one.

**Comment by matte-schwartz** [comment]
> Replying to https://github.com/ValveSoftware/Proton/issues/7348#issuecomment-1862003110

apologies for that! I've restarted Steam and my Deck several times and I keep getting `Proton: 1702923837 experimental-bleeding-edge-8.0-70057-20231218-p4bbc22-w0da56e-d1b31aa-v105b5b` as the latest version, so maybe the commit is still a little to bleeding edge for my Steam Client to recognize it. I'll continue testing it periodically over the next couple hours and will share any new logs if I manage to download the updated release.

**Comment by matte-schwartz** [comment]
okay, deleted the entire Experimental folder manually and it downloaded `Proton: 1702950110 experimental-bleeding-edge-8.0-70086-20231219-p301e0c-wfd53e0-d1b31aa-v105b5b`. 

This version appears to be much better! I was able to make it past the menu screen and intro without any workarounds whatsoever. The intro video sequence and title card that is supposed to play in the middle of Chapter 1 still got skipped, but the game continued on afterwards. I've made it all the way through Chapter 5, which is where many special fx start to get stacked on top of each other. No problems of any sort. 

It is now playable for the most part, just without video. Thank you for the prompt fixes for the hard game crashes!

Updated Proton log here: 
[steam-2052410.log](https://github.com/ValveSoftware/Proton/files/13710700/steam-2052410.log)

**Comment by R1kaB3rN** [comment]
Yes, I can confirm that updating to the latest bleeding edge Proton no longer results in crashing (yet) when either opening the backlog or progressing the script. Still **this is a workaround** until the developers properly fix the use after free bugs of the engine. Also, if the developers re-encode the in-game videos to media formats compatible with Proton then this game will be well on the path to being Steam Deck Verified

**Comment by R1kaB3rN** [comment]
Besides the above issues, the game just really drains the battery. Both system CPU/GPU utilization are unusually high which is pretty unacceptable IMO given the format of this game. On the Steam Deck, you can see this at the intro but just be sure the Performance Overlay Level is at least at 2. I don't know what techniques the engine is doing or if this is a result of the use after free bugs, but the game's developers should _really_ look into fixing this and optimizing the engine

**Comment by MuraPura** [comment]
Game appears to work perfectly on my system using the latest bleeding-edge. Thank you all so much!

**Comment by fallenguru** [comment]
@MuraPura, your post makes it sound like the game is fixed, but AFAIK the videos are still being (silently) skipped?

**Comment by matte-schwartz** [comment]
@fallenguru you are correct, videos still do not play. the intro sequence with the title theme gets totally skipped. The Proton update fixed the game crashing, not video playback.

**Comment by MuraPura** [comment]
> @MuraPura, your post makes it sound like the game is fixed, but AFAIK the videos are still being (silently) skipped?

Correct, videos still do not play. The reason I didn't mention that is because this is due to missing video codecs and not due to a programming bug on the Proton developers' fault. Unless valve is able to work out an agreement with the owners of those codecs, it's not something they are able to fix at the moment AFAIK. If the codecs are added, it works perfectly, but that likely won't be done on official Proton builds anywhere in the near future. I still should have mentioned the videos don't play in my previous post though, sorry. There are ways to add these codecs if you look around online, and unofficial builds that will likely be updated to include these crashing fixes soon, but nothing that the Proton dev team is legally able to do.

**Comment by R1kaB3rN** [comment]
> The reason I didn't mention that is because this is due to missing video codecs 

From the logs, the `wrong ELF class: ELFCLASS64`  Gstreamer warnings are due to a failure of loading a plugin, which left me to believe that it was because of a bug or an unimplemented feature in Wine. I believe those videos should have played because Proton already includes those Gstreamer plugins such as `navigationtest` as they're part of either the Base or Good plugin package. If this behavior is correct, then maybe a better error message should have been outputted?

@gofman do you have insight on this?

**Comment by guymanley** [comment]
Glad the game doesn't crash out and instead allows you to skip FMVs on experimental now. 

`<Workaround removed by moderator>`

**Comment by kisak-valve** [comment]
Hello @guymanley, the workaround you posted is legally problematic and has been removed.

**Comment by R1kaB3rN** [comment]
[Recently there was an update for this game](https://store.steampowered.com/news/app/2052410/view/3987441005292745504?l=english) and after some quick testing on stable (Proton 8.0-4), it seems that the use-after-free bug that was making this game unplayable was _silently_ fixed or there was _some_ attempt at fixing it. Though, I haven't looked into the API trace to confirm

You can see this by just playing the intro again

Also, the in-game video for the intro is still skipped

**Comment by amayra** [comment]
do i need to install LAV filters ?

**Comment by YoreonX** [comment]
If you use Hotbar Menu set to Joystick as a Controller Config on Linux, specifically with this game.

None of the Hotbar commands will input, they do work on Windows, but with this game specifically they do not work.

For example, L = Load, if you set a Hotbar command to "L" the command simply will not input on Linux which is very strange...

Again tested on Windows and it works as intended.
